### ISSUE TITLE
[component-class-suffix] Allow configuration for component class suffix rule to support both Angular style guide and file-level consistency

### ISSUE THREAD

**Issue by saiyaff** [angular-eslint/angular-eslint#2497]
**Description and reproduction of the issue**

This also could be applied to the other such as [directive-class-suffix]

Error:

> /path/src/app/app.ts
>   9:14  error  Component class names should end with one of these suffixes: "Component" (https://angular.dev/style-guide#style-02-03)  @angular-eslint/component-class-suffix
> 

The old Angular style guide has recommended that component class names should end with Component (e.g., UserProfileComponent). The @angular-eslint/component-class-suffix rule enforces this, which is great for consistency across a codebase.

However, the new style guide states:

> "Whenever you encounter a situation in which these rules contradict the style of a particular file, prioritize maintaining consistency within a file. Mixing different style conventions in a single file creates more confusion than diverging from the recommendations in this guide."

Currently, the linter does not provide a way to allow both the recommended suffix and file-level consistency. This means that in legacy codebases or files with established naming conventions, we are forced to either refactor everything or disable the rule, which reduces the effectiveness of linting.

Please consider adding a configuration option to @angular-eslint/component-class-suffix that allows specifying multiple valid suffixes, or a way to relax the rule for files that are internally consistent but do not follow the global convention.

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@angular-eslint/eslint-plugin` | `19.7.0` |
| `@typescript-eslint/parser`     | `NA` |
| `ESLint`                        | `9.9.1` |
| `node`                          | `22.15.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 20.0.0
Node: 22.15.0
Package Manager: npm 11.3.0
OS: darwin arm64

Angular: 20.0.0
... animations, build, cli, common, compiler, compiler-cli, core
... forms, language-service, platform-browser
... platform-browser-dynamic, router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.2000.0
@angular-devkit/core         20.0.0 (cli-only)
@angular-devkit/schematics   20.0.0
@angular/cdk                 20.0.1
@angular/material            20.0.1
@schematics/angular          20.0.0
rxjs                         7.8.2
typescript                   5.8.3
zone.js                      0.15.0
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

[.eslintrc.json](https://github.com/user-attachments/files/20558253/default.eslintrc.json)

**Comment by JamesHenry** [comment]
Thanks for the report @saiyaff and I'm sorry for the long delay. A few important things have changed since this was filed:

- As of angular-eslint `20.1.0`, `component-class-suffix` and `directive-class-suffix` are no longer part of the `recommended` config. They are opt-in only, so new Angular 20 and later projects are not affected unless you enable them.
- Both rules already accept a `suffixes` array if you want to allow more than one suffix.

For the "keep the suffix in existing code, drop it in new code" case, a per-file consistency option would rarely fire because Angular files almost always contain one component (or none). Flat config handles this better: enable the rule only for the paths that should keep the convention, for example

```js
{
  files: ['src/app/legacy/**/*.ts'],
  rules: { '@angular-eslint/component-class-suffix': 'error' },
}
```

and leave it off everywhere else. Closing as not planned, but happy to reopen if there is a concrete case the override approach cannot cover.
